packages/rs-drive-proof-verifier/src/from_request.rs

```diff
@@ -0,0 +1,449 @@
+            GrpcResultType::VoteTally => DriveResultType::VoteTally,
+        })
+    }
+    fn try_to_request(
+        &self,
+    ) -> Result<get_contested_resource_vote_state_request_v0::ResultType, Error> {
+        use get_contested_resource_vote_state_request_v0::ResultType as GrpcResultType;
+        use ContestedDocumentVotePollDriveQueryResultType as DriveResultType;
+
+        Ok(match self {
+            DriveResultType::Documents => GrpcResultType::Documents,
+            DriveResultType::DocumentsAndVoteTally => GrpcResultType::DocumentsAndVoteTally,
+            DriveResultType::IdentityIdsOnly => GrpcResultType::IdentityIdsOnly,
+            DriveResultType::VoteTally => GrpcResultType::VoteTally,
+        })
+    }
+}
+
+impl TryFromRequest<GetContestedResourceVoteStateRequest> for ContestedDocumentVotePollDriveQuery {
+    fn try_from_request(grpc_request: GetContestedResourceVoteStateRequest) -> Result<Self, Error> {
+        let result = match grpc_request.version.ok_or(Error::EmptyVersion)? {
+            get_contested_resource_vote_state_request::Version::V0(v) => {
+                ContestedDocumentVotePollDriveQuery {
+                    limit: v.count.map(|v| v as u16),
+                    vote_poll: ContestedDocumentResourceVotePoll {
+                        contract_id: Identifier::from_bytes(&v.contract_id).map_err(|e| {
+                            Error::RequestError {
+                                error: format!("cannot decode contract id: {}", e),
+                            }
+                        })?,
+                        document_type_name: v.document_type_name.clone(),
+                        index_name: v.index_name.clone(),
+                        index_values: bincode_decode_values(v.index_values.iter())?,
+                    },
+                    result_type:  match v.result_type() {
+                        get_contested_resource_vote_state_request_v0::ResultType::Documents => {
+                            ContestedDocumentVotePollDriveQueryResultType::Documents
+                        }
+                        get_contested_resource_vote_state_request_v0::ResultType::DocumentsAndVoteTally => {
+                            ContestedDocumentVotePollDriveQueryResultType::DocumentsAndVoteTally
+                        }
+                        get_contested_resource_vote_state_request_v0::ResultType::IdentityIdsOnly => {
+                            ContestedDocumentVotePollDriveQueryResultType::IdentityIdsOnly
+                        }
+                        get_contested_resource_vote_state_request_v0::ResultType::VoteTally => {
+                            ContestedDocumentVotePollDriveQueryResultType::VoteTally
+                        }
+                    },
+                    start_at: v
+                        .start_at_identifier_info
+                        .map(|v| to_bytes32(&v.start_identifier).map(|id| (id, v.start_identifier_included)))
```

Comment:
no need for to_bytes32. you can just to v.start_identifier.to_buffer

Reply:
pub start_identifier: ::prost::alloc::vec::Vec<u8>,

Vec<u8>  does not have to_buffer().